src/mesh/region.cc

```diff
@@ -568,3 +407,51 @@ Region RegionDB::find_by_dimid(DimIDIter it_id, unsigned int id, const std::stri
 
     return r_id;
 }
+
+void RegionDB::print_region_table(ostream& stream) const {
+	ASSERT(closed_, "RegionDB not closed yet.\n");
+
+	// print header
+	stream << endl << "----------- Table of all regions: -----------" << endl;
+	stream << std::setfill(' ') << setw(6) << "id" << " dim label" << setw(12) << "" << "contains regions" << endl;
+	// print data
+	for (RegionSetTable::const_iterator it = sets_.begin(); it != sets_.end(); ++it) { // iterates through RegionSets
+		unsigned int reg_id = RegionIdx::undefined;
+		string rset_label = it->first;
+		LabelIter label_it = region_table_.get<Label>().find(rset_label);
+		if ( label_it != region_table_.get<Label>().end() ) { // checks if Region with same label as RegionSet exists
```

Comment:
Result of set opeartions is just a region set i.e. non-elementary region and should have same format as all other sets. The point of my comment is, that we should distinguish just two kind of sets:
1) single region sets with name equal to region should be printed as elementary regions with ID and dim
2) other sets should be printed as list of region labels, here I suggest enclosing the list into brackets, e.g.

.region_xy 		boundary 	id: 101	dim: 2
region_yz 		bulk		 	id: 102	dim: 1
set				[ .region_xy, region_yz ]